INDI/INDI/indiserver.c

```diff
@@ -1948,150 +1958,211 @@ static int readFromDriver(DvrInfo *dp)
     return (shutany ? -1 : 0);
 }
 
-/* service traffic from clients and drivers */
-static void indiRun(void)
-{
-    fd_set rs, ws;
-    int maxfd = 0;
-    int i, s;
-
-    /* init with no writers or readers */
-    FD_ZERO(&ws);
-    FD_ZERO(&rs);
+/****************************************************/
+/* Setup routines for select(...) call in indiRun() */
+/****************************************************/
```

Comment:
Simplify indiRun by refactoring and abstracting the different types of select(2) setup and handling it does.